init_node.sh

```diff
@@ -1,88 +1,155 @@
+    CHAIN_DIR="${CHAIN_DIR_PREFIX}${i}"
+
+    # Add dir for chain, exit if error
+    if ! mkdir -p ${CHAIN_DIR} 2>/dev/null; then
+      echo "Failed to create chain folder(${CHAIN_DIR}). Aborting..."
+      exit 1
+    fi
+
+    # Initialize configuration files and genesis file
+    # moniker is the name of your node
+    MONIKER="${MONIKER_PREFIX}${i}"
+    redirect ${BINARY} init ${MONIKER} --home ${CHAIN_DIR} --chain-id ${CHAIN_ID}
+
+    # Create N genesis account(with mnemonic), so N-th chain's N account is same with M-th chain's
+    for ((j = 0; j < N; j++))
+      do
+        VALIDATOR="${VALIDATOR_PREFIX}${j}"
+        ${BINARY} keys add ${VALIDATOR} ${KEYRING} --home ${CHAIN_DIR} --recover --account ${j} --output json <<< ${TEST_MNEMONIC} >> ${CHAIN_DIR}/validator_seed.json 2> /dev/null
+        redirect ${BINARY} add-genesis-account $(${BINARY} --home ${CHAIN_DIR} keys ${KEYRING} show ${VALIDATOR} -a) ${COINS} --home ${CHAIN_DIR}
+      done
+
+    # Make gentx file and move it to GENTXS folder
+    VALIDATOR="${VALIDATOR_PREFIX}${i}"
+    redirect ${BINARY} gentx ${VALIDATOR} ${DELEGATE} ${KEYRING} --home ${CHAIN_DIR} --chain-id ${CHAIN_ID}
+    mv "${CHAIN_DIR}/config/gentx/$(ls ${CHAIN_DIR}/config/gentx | grep .json)" "${GENTXS_DIR}/${MONIKER}.json"
+    rm -r "${CHAIN_DIR}/config/gentx"
+  done
+
+SRC_GENESIS_FIlE="${CHAIN_DIR_PREFIX}0/config/genesis.json"
+RPC_PORT=26657
+P2P_PORT=26656
+PROF_PORT=6060
+GRPC_PORT=9090
+
+# Set genesis file and config(port, peer, ...)
+CHAIN_0_DIR="${CHAIN_DIR_PREFIX}0"
+for ((i = 0; i < N; i++))
+  do
+    CHAIN_DIR="${CHAIN_DIR_PREFIX}${i}"
+
+    # Set genesis file of 0-th chain dir and copy to other chains
+    # If we call collect-gentxs at each chains, genesis_time values can be different.
+    if [ ${i} -eq 0 ]; then
+      redirect ${BINARY} collect-gentxs --home ${CHAIN_DIR} --gentx-dir ${GENTXS_DIR}
+    else
+      cp ${SRC_GENESIS_FIlE} "${CHAIN_DIR}/config"
+    fi
+
+    MONIKER="${MONIKER_PREFIX}${i}"
+    MEMO=`sed 's/"//g' <<< \`cat ${GENTXS_DIR}/${MONIKER}.json | jq '.body.memo'\``
+    MEMO_SPLIT=(`echo ${MEMO} | tr ":" "\n"`)
+
+    # Set proper defaults and change ports (use a different sed for Mac or Linux)
+    if [ "`uname`" == 'linux' ]; then
+      sed -i 's#"tcp://127.0.0.1:26657"#"tcp://0.0.0.0:'"${RPC_PORT}"'"#g' ${CHAIN_DIR}/config/config.toml
+      sed -i 's#"tcp://0.0.0.0:26656"#"tcp://0.0.0.0:'"${P2P_PORT}"'"#g' ${CHAIN_DIR}/config/config.toml
+      sed -i 's#"localhost:6060"#"localhost:'"${P2P_PORT}"'"#g' ${CHAIN_DIR}/config/config.toml
+      sed -i 's/timeout_commit = "5s"/timeout_commit = "1s"/g' ${CHAIN_DIR}/config/config.toml
+      sed -i 's/timeout_propose = "3s"/timeout_propose = "1s"/g' ${CHAIN_DIR}/config/config.toml
+      sed -i 's/index_all_keys = false/index_all_keys = true/g' ${CHAIN_DIR}/config/config.toml
+      sed -i 's/persistent_peers = ""/persistent_peers = "${PERSISTANT_PEERS}"/g' ${CHAIN_DIR}/config/config.toml
```

Comment:
Please use the same quotation trick as macOS's (PERSISTANT_PEERS variable is not evaluated).

Reply:
persistent_peer line is useless, I'll remove it